src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -1267,6 +1269,61 @@ internal static void TagActivityWithOrchestrationStatus(OrchestrationRuntimeStat
                 activity.AddTag("DurableFunctionsRuntimeStatus", statusStr);
             }
         }
+
+        internal IScaleMonitor GetScaleMonitor(string functionId, FunctionName functionName, string storageConnectionString)
+        {
+            if (this.defaultDurabilityProvider.TryGetScaleMonitor(
+                functionId,
+                functionName.Name,
+                this.Options.HubName,
+                storageConnectionString,
+                this.logger,

```

Comment:
I feel like it's not a good idea to pass the `ILogger` used by the extension to the 3rd party durability provider. That basically gives it the ability to emit logs that look like they are coming from the extension when really they are not. Instead, I would prefer we either pass in the `ILoggerFactory` so that the durability provider can generate it's own `ILogger` with its own category, OR create a new `ILogger` with a category name we define, based on the name of the durability provider.

Thoughts?

Reply:
That's fair. In fact, we may not even need this as a parameter, because the IDurabilityProviderFactory implementation can just access ILoggerFactory by adding it as a parameter on its constructor.

Reply:
That makes sense. We should keep it separate. Taking Azure Storage as an example, `AzureStorageDurabilityProviderFactory` has an `ILoggerFactory` constructor parameter. Is it fine to create an `ILogger` in `GetDurabilityProvider()` and then pass that `ILogger` when creating an `AzureStorageDurabilityProvider`?

Example in `GetDurabilityProvider()` in `AzureStorageDurabilityProviderFactory`:
```
ILogger logger = this.loggerFactory.CreateLogger("AzureStorage");
this.defaultStorageProvider = new AzureStorageDurabilityProvider(
    defaultService,
    this.defaultConnectionName,
    this.azureStorageOptions,
    logger);
```

Then, `TryGetScaleMonitor()` would use the `ILogger` that's passed into `AzureStorageDurabilityProvider` and the `ILogger` parameter in `TryGetScaleMonitor()` can be removed.

Reply:
Yeah, that makes the most sense I think.